### PR TITLE
Fix duplicated newlines in GDScript bytecode

### LINKED ISSUE #26789: Off by one error on breakpoints in combination with for loops and comments
**Godot version:**

Godot 3.1 b376273, debug & release_debug builds

**OS/device including version:**
Linux Mint 19 x64

**Issue description:**
When outputting messages via `print()` in nested `for` cycles, not all lines are being outputted if a breakpoint is set AFTER the corresponding output code:

image

The code above with breakpoint set outputs only 5 lines of messages instead of expected 8. So it seems that the second cycle didn't finish it's whole set of iterations.

This is valid for engine output pane and command line terminal as well.
And such behavior is quite misleading and could have some deeper roots in debugger/gdscirp behavior (???).

Interesting detail, though, is that if one puts additional `print` statement before the breakpoint, all prior calls became valid, including the new one. ( See the commented `print()` before the `pass`)

Also, not only the breakpoints set on `pass` are displaying this behavior, but the ones in more complex code examples do as well.

**Steps to reproduce:**
Write two nested cycles with `print` output in each.
Set a breakpoint after the second cycle.
Run project in the debug mode.
See that not all of the expected output is present in corresponding editor pane and in the command line terminal.

**Minimal reproduction project:**
[godot_double_cycle.zip](https://github.com/godotengine/godot/files/2945894/godot_double_cycle.zip)

### PR DESCRIPTION
**Pull request by bojidar-bg**
Fixes #26789.

### CODE REVIEW

File: modules/gdscript/gdscript_parser.cpp [hunk 1]
```diff
@@ -2741,6 +2741,8 @@ void GDScriptParser::_parse_block(BlockNode *p_block, bool p_static) {
 			} break;
 			case GDScriptTokenizer::TK_NEWLINE: {
 
+				int line = tokenizer->get_token_line();
```
Comment (aaronfranke) [inline modules/gdscript/gdscript_parser.cpp]: Is it worth creating a variable for something that's only used once?

**Review by vnen — APPROVED** [review]
LGTM

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.1.1.

**Pull request merged**